src/components/loadData.js

```diff
@@ -31,5 +31,18 @@ export const loadData = (setPosts, setError) => {
     return currentPosts.then((previousPosts) => posts()
       .then((newPosts) => [...previousPosts, ...newPosts]));
   }, Promise.resolve([]))
-  .then(setPosts).catch(setError);
-}
+  .then((posts) => {
+    setPosts(posts);
+    loadComments(posts).then(() => setPosts([...posts]));
```

Comment:
We should avoid mutating data that we have saved in the component state. The reason for that is that React depends on object identity to know something change, so if you mutate things that are already in state React won't know about it. I'm assuming you did `setPosts([...posts])` there because when you used just `setPosts(posts)` it was not re-rendering, right? This is because spread (the `...` operator) will create a new array which React will know it's different from the one set in the previous line, but it's still error prone because the _content_ of the array keeps the same (spread won't deep clone the array) so we're still mutating something that is saved to the state.

A suggestion to fix it is returning a new array of posts containing their comments (we could use `.map` instead of `forEach`) and instead of doing `post.comments = value` we create a new object for each mapping iteration. What do you think?